build: fix, upgrade dependencies

joemfb opened the pull request:
This PR upgrades nixpkgs to latest master, which results in the following dependency upgrades:

```
gmp:     6.2.0  -> 6.2.1
sigsegv: 2.12   -> 2.14
openssl: 1.1.1g -> 1.1.1n
libuv:   1.40.0 -> 1.43.0
lmdb:    0.9.25 -> 0.9.29
curl:    7.72.0 -> 7.81.0
```

Most notably, this change upgrades secp256k1 (which is unversioned), and includes breaking changes to its API for schnorr signatures, which necessitates changes to urcrypt. It also upgrades LLVM from version 7 to 11 (which took about 16 hours, since the darwin `stdenv` embeds it three (3) times).

#5561 showed up again, and is apparently not M1 specific; it occurs on macos x86_64 as well (and is probably just a clang bug).

I was unable to get `herb` to build inside nix (issues with `pytest-forked` and python2), so I removed `herb` from our test suite. The pill builders are now broken, but it looks like they've been unused since at least the software-distribution upgrade in arvo.

Finally, I've tweaked the github-actions build triggers, and specifically tried to prevent "releasing" the docker image on pre-release tags. I also tried to embed the statically-linked urbit binary in the docker image, but the obvious `--arg enableStatic true` led to re-bootstrapping the universe. I'm open to suggestions ...

/cc @frodwith (re:urcrypt)

mrdomino commented:
schnorr changes look good. Apparently they changed the API again since the last merge to `nixpkgs` master, so at some point we will have to use `…sign32` instead of `…sign`, but that's neither here nor there.

joemfb commented on .github/workflows/release-docker.yml:
```diff
@@ -3,7 +3,7 @@ name: release-docker
   release: null
   push:
-    tags: ['urbit-v*']
+    tags: 'urbit-v[0-9]+.[0-9]+'
```
*semver delenda est*

joemfb commented:
I went ahead and fixed up the pill builders as well. Replaced `herb` with `curl`, switched `%home` to `%base`, &c. (There are permissions problems with the solid builder on my mac -- not sure what's going on, but it's in a better state than before ...)

leowini commented:
@joemfb Were you able to get the solid builder to work on Mac? I think I'm getting the same permissions issue on line 89 of solid.sh: =chmod: changing permissions of './pier/stage': Operation not permitted.

leowini commented:
I have a feeling it has to do with this: [](https://github.com/NixOS/nix/issues/4640)
My /nix is owned by root. I'll try to reinstall nix.

leowini commented:
I have a feeling it has to do with this: https://github.com/NixOS/nix/issues/4640
My /nix is owned by root. I'll try to reinstall and see if that fixes it.
